cmd/publishing-bot/publisher.go

```diff
@@ -71,472 +42,45 @@ type PublisherMunger struct {
-					{repo: "apiserver", branch: "master"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.6 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.6", dir: "staging/src/k8s.io/sample-apiserver"},
-				dst: coordinate{repo: "sample-apiserver", branch: "release-1.6", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.6"},
-					{repo: "client-go", branch: "release-3.0"},
-					{repo: "apiserver", branch: "release-1.6"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.7 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.7", dir: "staging/src/k8s.io/sample-apiserver"},
-				dst: coordinate{repo: "sample-apiserver", branch: "release-1.7", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.7"},
-					{repo: "client-go", branch: "release-4.0"},
-					{repo: "apiserver", branch: "release-1.7"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.8 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.8", dir: "staging/src/k8s.io/sample-apiserver"},
-				dst: coordinate{repo: "sample-apiserver", branch: "release-1.8", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.8"},
-					{repo: "api", branch: "release-1.8"},
-					{repo: "client-go", branch: "release-5.0"},
-					{repo: "apiserver", branch: "release-1.8"},
-				},
-			},
-			{
-				// rule for the sample-apiserver 1.9 branch
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.9", dir: "staging/src/k8s.io/sample-apiserver"},
-				dst: coordinate{repo: "sample-apiserver", branch: "release-1.9", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.9"},
-					{repo: "api", branch: "release-1.9"},
-					{repo: "client-go", branch: "release-6.0"},
-					{repo: "apiserver", branch: "release-1.9"},
-				},
-			},
-		},
-		publishScript: "/publish_scripts/publish_sample_apiserver.sh",
-	}
-
-	sampleController := repoRules{
-		skipped: false,
-		dstRepo: "sample-controller",
-		srcToDst: []branchRule{
-			{
-				src: coordinate{repo: config.SourceRepo, branch: "master", dir: "staging/src/k8s.io/sample-controller"},
-				dst: coordinate{repo: "sample-controller", branch: "master", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "master"},
-					{repo: "api", branch: "master"},
-					{repo: "client-go", branch: "master"},
-				},
-			},
-			{
-				src: coordinate{repo: config.SourceRepo, branch: "release-1.9", dir: "staging/src/k8s.io/sample-controller"},
-				dst: coordinate{repo: "sample-controller", branch: "release-1.9", dir: "./"},
-				deps: []coordinate{
-					{repo: "apimachinery", branch: "release-1.9"},
-					{repo: "api", branch: "release-1.9"},
-					{repo: "client-go", branch: "release-6.0"},
-				},
-			},
-		},
-		publishScript: "/publish_scripts/publish_sample_controller.sh",
+	if err := p.readRepositoryRules(cfg); err != nil {
```

Comment:
We call this here and in updateSourceRepo. Feels strange.

If we read it from the source repo, we have to update the source repo first before read the rules. Do we have the source repo name without the rules? From the config map?